packages/palette/src/elements/Modal/index.tsx

```diff
@@ -0,0 +1,238 @@
+            opacity: 0,
+          })
+          onClose()
+        },
+      })
+    } else {
+      setVisibleContent(content)
+      setSpringModalAnimation({
+        transform: "translate(-50%, -50%) translateY(0vh)",
+        opacity: 1,
+        onRest: null,
+      })
+    }
+  }, [show])
+
+  // Listens for props to change and fades in new content
+  useEffect(() => {
+    if (
+      (previousChildren && children !== previousChildren) ||
+      (previousFixedButton &&
+        FixedButton &&
+        FixedButton.props.children !== previousFixedButton.props.children) ||
+      (previousTitle && title !== previousTitle) ||
+      (previousHasLogo && hasLogo !== previousHasLogo)
+    ) {
+      setSpringContentAnimation({
+        opacity: 0,
+        onRest: () => updateVisibleContent(),
+      })
+    }
+  }, [hasLogo, title, FixedButton, children])
+
+  const content = () => {
+    return (
+      <AnimatedView style={contentAnimation}>
+        <ModalScrollContent FixedButton={FixedButton}>
+          {hasLogo && (
+            <>
+              <Flex justifyContent="center">
+                <Logo />
+              </Flex>
+              <Spacer mb={space(1)} />
+            </>
+          )}
+          {title && (
+            <>
+              <Flex justifyContent="center">
+                <Serif size="5t" textAlign="center" color="black100">
+                  {title}
+                </Serif>
+              </Flex>
+              <Spacer mb={space(1)} />
+            </>
+          )}
+          {children}
+        </ModalScrollContent>
+        {FixedButton && <FixedButtonWrapper>{FixedButton}</FixedButtonWrapper>}
+      </AnimatedView>
+    )
+  }
+
+  const updateVisibleContent = () => {
+    setChildrenState(children)
+    setTitleState(title)
+    setFixedButtonState(FixedButton)
+    setHasLogoState(hasLogo)
+    // Replaces content
+    setVisibleContent(content)
+    // Fades in new content
+    setSpringContentAnimation({
+      opacity: 1,
+      onRest: null,
+    })
+  }
+
+  return (
+    <ModalOuterWrapper show={show}>
+      <ModalWrapper>
+        <ModalElement style={modalAnimation} isWide={isWide} show={show}>
+          <CloseIconWrapper onClick={() => onClose()}>
+            <CloseIcon fill="black60" />
+          </CloseIconWrapper>
+          {visibleContent}
+        </ModalElement>
+      </ModalWrapper>
+    </ModalOuterWrapper>
+  )
+}
+
+const FixedButtonWrapper = styled(Box)`
+  position: fixed;
+  bottom: 0;
+  left: 0;
+  width: 100%;
+  padding-left: ${space(4)}px;
+  padding-right: ${space(4)}px;
+  padding-top: ${space(2)}px;
+  padding-bottom: ${space(3)}px;
+  background-color: ${color("white100")};
+  border-top: 1px solid ${color("black10")};
+`
+
+const ModalOuterWrapper = styled(Box)<TransitionElementProps>`
+  position: fixed;
+  top: 0;
+  left: 0;
+  z-index: 100;
+  width: 100vw;
+  height: 100vh;
+  background-color: rgba(229, 229, 229, 0.5);
+  transition: opacity 250ms ease;
```

Comment:
Why mix react-spring + CSS animation?

Reply:
I'd like to use `react-spring` sparingly and therefore it's only used when an animation finished callback is necessary and only on a single element.